### PR TITLE
Cockpit 237

### PR DESCRIPTION
**Pull request by marusak**
Preview https://marusak.dev/cockpit-project.github.io/blog/cockpit-237.html

We still need at least one or two PRs to land, but opening for general review.

### CODE REVIEW

File: _posts/2021-02-03-cockpit-237.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.
+
+Here are the release notes from Cockpit version 237.
+
+## Updates: Restyling as preparation for upcoming new fatures
+
+Updates page went through some design polishing as preparation for new upcoming features as support
+of [tracer](https://tracer.readthedocs.io/en/latest/) and [kpatch](https://github.com/dynup/kpatch).
```
Comment (martinpitt) [inline _posts/2021-02-03-cockpit-237.md:18]: Let's please not announce vaporware.  I'd stop after "for upcoming features.". If you do want to keep it, "features, like support for tracer and kpatch".

File: _posts/2021-02-03-cockpit-237.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+... screenshot ... (TODO we need to fix at least the page filling and possibly also dropping title panel).
+
+## Shell: SSH connections to remote machines are only opened when necessary
+
+Cockpit remembers all remote hosts of a session and used to immediately connect to all of them
+when starting the next session. Now a SSH connection is only opened when navigating to the remote
+machine for the first time.
```
Comment (martinpitt) [inline _posts/2021-02-03-cockpit-237.md:26]: We should give a rationale here: It restricts possible security issues, as any connected host can do *anything* to any other connected host. (With better verbiage, obviously)

File: _posts/2021-02-03-cockpit-237.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Internal CI improvements
+
+Team has been focusing in the last two weeks heavily of stabilizing and improving our CI. These
```
Comment (martinpitt) [inline _posts/2021-02-03-cockpit-237.md:30]: In the last two weeks, the team has been focusing heavily on stabilizing.."

File: _posts/2021-02-03-cockpit-237.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+improvements include work on self-testing CI infrastructure as well as implementation of [error
+budget](https://sre.google/sre-book/embracing-risk/). These changes don't have a user-visible effect,
+but will make Cockpit easier to develop in the future. Also we continued working on tests and build system improvements.
+On top all of that there is a bug fix in websocket thanks to contribution from mehmooda. _Thanks, mehmooda!_
```
Comment (martinpitt) [inline _posts/2021-02-03-cockpit-237.md:34]: We don't usually mention bug fixes, but if we do, there should be some details about it.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks!

File: _posts/2021-02-03-cockpit-237.md (hunk 4 above)
Comment (marusak) [reply _posts/2021-02-03-cockpit-237.md:34]: Right, we just wanted to mention external contributor. @allisonkarlitskaya

File: _posts/2021-02-03-cockpit-237.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+summary: Cockpit with restyled updates page
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.
+
+Here are the release notes from Cockpit version 237.
+
+## Updates: Restyling as preparation for upcoming new fatures
+
+Updates page went through some design polishing as preparation for new upcoming features as support
```
Comment (allisonkarlitskaya) [inline _posts/2021-02-03-cockpit-237.md:17]:
```suggestion
The updates page went through some design polishing as preparation for upcoming features such as support
```

File: _posts/2021-02-03-cockpit-237.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+slug: cockpit-237
+category: release
+summary: Cockpit with restyled updates page
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.
+
+Here are the release notes from Cockpit version 237.
+
+## Updates: Restyling as preparation for upcoming new fatures
```
Comment (allisonkarlitskaya) [inline _posts/2021-02-03-cockpit-237.md:15]:
```suggestion
## Updates: Restyling as preparation for upcoming features
```

File: _posts/2021-02-03-cockpit-237.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+Updates page went through some design polishing as preparation for new upcoming features as support
+of [tracer](https://tracer.readthedocs.io/en/latest/) and [kpatch](https://github.com/dynup/kpatch).
+
+... screenshot ... (TODO we need to fix at least the page filling and possibly also dropping title panel).
+
+## Shell: SSH connections to remote machines are only opened when necessary
+
+Cockpit remembers all remote hosts of a session and used to immediately connect to all of them
```
Comment (allisonkarlitskaya) [inline _posts/2021-02-03-cockpit-237.md:24]:
```suggestion
Cockpit remembers all remote hosts of a session and would previously connect to all of them
```

File: _posts/2021-02-03-cockpit-237.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+when starting the next session. Now a SSH connection is only opened when navigating to the remote
```
Comment (allisonkarlitskaya) [inline _posts/2021-02-03-cockpit-237.md:25]:
```suggestion
when starting the next session. Now an SSH connection is only opened when navigating to the remote
```

File: _posts/2021-02-03-cockpit-237.md (hunk 4 above)
Comment (allisonkarlitskaya) [reply _posts/2021-02-03-cockpit-237.md:34]: Maybe include full name?  It's listed on the GitHub profile.

Comment (allisonkarlitskaya) [inline _posts/2021-02-03-cockpit-237.md:34]:
```suggestion
On top all of that there is a bug fix in websocket thanks to a contribution from mehmooda. _Thanks, mehmooda!_
```

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

File: _posts/2021-02-03-cockpit-237.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+The updates page went through some design polishing in preparation for upcoming features.
+
+![Restyled updates page](/images/updates-restyled.png)
+
+## Shell: SSH connections to remote machines are only opened when necessary
+
+Cockpit remembers all remote hosts of a session and would previously connect to all of them
+when starting the next session. Now an SSH connection is only opened when navigating to the remote
+machine for the first time to restricts possible security issues, as connected host can
+interacted with each other.
```
Comment (martinpitt) [inline _posts/2021-02-03-cockpit-237.md:26]: Runaway sentence, and some grammar issues.

```suggestion
machine for the first time. This restricts possible security issues, as connected hosts can
interact with each other.
```

File: _posts/2021-02-03-cockpit-237.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+## Internal CI improvements
+
+In the last two weeks, the team has been focusing heavily on stabilizing and improving our CI. These
+improvements include work on self-testing CI infrastructure as well as implementation of [error
+budget](https://sre.google/sre-book/embracing-risk/). These changes don't have a user-visible effect,
```
Comment (martinpitt) [inline _posts/2021-02-03-cockpit-237.md:32]:
```suggestion
improvements include work on self-testing CI infrastructure as well as [defining](https://github.com/cockpit-project/cockpit/wiki/DevelopmentPrinciples#our-testsci-error-budget) and [implementing](https://images-frontdoor.apps.ocp.ci.centos.org/tests.html) an [error budget](https://sre.google/sre-book/embracing-risk/). These changes don't have a user-visible effect,
```

File: _posts/2021-02-03-cockpit-237.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+but will make Cockpit easier to develop in the future. Also we continued working on tests and build system improvements.
+On top all of that there is a bug fix in websocket thanks to a contribution from Asad Mehmood. _Thanks, mehmooda!_
```
Comment (martinpitt) [inline _posts/2021-02-03-cockpit-237.md:34]:
```suggestion
On top all of that there is a bug fix in websocket; _thanks to Asad Mehmood for that contribution!_
```

**Review by martinpitt — CHANGES_REQUESTED** [review]

**Review by allisonkarlitskaya — APPROVED** [review]

**Comment by martinpitt** [comment]
LGTM, just needs new bodhi URLs.

**Review by martinpitt — APPROVED** [review]
Dakujem!

**Comment by marusak** [comment]
links added

**Pull request merged**